lib/ZnapZend/ZFS.pm

```diff
@@ -289,6 +314,15 @@ sub sendRecvSnapshots {
     my $recvOpt = $self->recvu ? '-uF' : '-F';
     my $remote;
     my $mbufferPort;
+    my $dstDataSetExists = $self->dataSetExists($dstDataSet);
+
+    #attemp a creation of the dataset when it doesn't exist
+    $dstDataSetExists = $self->createDataSet($dstDataSet) if $self->autoCreation && !$dstDataSetExists;
+
+    #check if the dstDataSet exist on the destination (maybe after a creation)
+    !$dstDataSetExists
+        and Mojo::Exception->throw('ERROR: dstDataSet does not exist on dest host');
+
```

Comment:
to preserve present functionality, you would first check for autoCreation and then check for the datasets existance and then create it. since create will already throw an exception if it fails, there is no need todo anything anymore here.